packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -708,14 +739,26 @@ class MessageChannel {
     this.channel = channel.port1
   }
 
-  public dispatchBuffered() {
+  public dispatchBuffered({ lifecycle }: { lifecycle?: Lifecycle } = {}) {
     console.assert(this.channel != null, 'channel is not setup')
 
-    this.bufferedMessages.forEach(([message, transferables]) => {
+    let messages: BufferedMessages
+
+    switch (lifecycle) {
+      case Lifecycle.AFTER_INITIALIZED:
+        messages = this.bufferedMessagesAfterInitialized
+        break
+
+      default:
+        messages = this.bufferedMessages
+        break
+    }
+
+    messages.forEach(([message, transferables]) => {
       this.channel?.postMessage(message, transferables ?? [])
     })
 
-    this.bufferedMessages = []
+    messages = []
```

Comment:
This won't reset buffered messages. I'd change the `forEach` loop above to something like this (untested):

```typescript
    while (messages.length > 0) {
      const [message, transferables] = messages.shift()
      this.channel?.postMessage(message, transferables ?? [])
    }
```